internal/services/recoveryservices/site_recovery_replicated_vm_resource_test.go

```diff
@@ -24,6 +26,10 @@ func TestAccSiteRecoveryReplicatedVm_basic(t *testing.T) {
 			Config: r.basic(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
+				// A temporary disk with name `ms-asr-*` is created and deleted automatically by recovery service when creating the replication.
+				// After this temporary disk is deleted, its entry still remains in the resource list of the resource group for some time, causing the deletion of the resource group to fail.
+				// Delete the temporary disk explicitly as a workaround
+				data.CheckWithClientForResource(r.deleteTempDisk(), "azurerm_resource_group.test"),
```

Comment:
should we maybe be doing this in the resource's delete function so it doesn't fail for users?

Reply:
We are not able to get the exact name of the generated disk because its named as `ms-asr-<random guid>`, here I just deleted the first one I found in that resource group. If we do this to the delete function, we could potentially delete a disk created by users themselves.
This temp disk is created and removed at the creation time of this resource, and its entry in the resource group will be cleaned up automatically in few hours. I guess it's fine if we don't touch it in the delete?

Reply:
but for users this resource won't properly delete? can you reach out to the service team about this as it sounds like a bug on their side?

Reply:
I've opened an issue at Azure Spec repo https://github.com/Azure/azure-rest-api-specs/issues/18993 describing the issue we see to track.
Create/Delete of `site_recovery_replicated_vm` itself is not affected by this temp disk. This issue affects the deletion of the resource group only. If a user is testing the deleting scenarios shortly after the creation is done (about 30 to 60 minutes in my test), this issue will be hit. If the deletion happens not very soon, the disk entry in the resource list will already be cleaned up by the service.
Before the service team looking into the issue, I'm considering using this as a workaround for our current test failure and #15783 due to this dangling disk starting from 3.0.

Reply:
This would affect users as well if they want to delete the resource group soon after creating the resource, therefor it seems to very much be a bug in the service. as there been any updates from the service team?

Reply:
there seems no update in the opened issue, I'll check with service side internally for an update. Meanwhile, is it ok to set `prevent_deletion_if_contains_resources` to `false` instead to unblock the test cases?

Reply:
Its still a bug in the API/service that needs to be fixed, we should not be hacking the test to pass

Reply:
Got it. We can then close this pr since it is only adding the workaround. After the upstream issue is resolved, the test case will pass.

Reply:
I'm going to leave it open for tracking for now